scripts/release.ts

```diff
@@ -93,71 +105,156 @@ function init(args) {
-    new URL("release-pull-description.md", import.meta.url)
-  );
-
-  const temporaryDir = mkdtempSync(join(tmpdir(), "pull-request-"));
-  const temporaryBodyFile = join(temporaryDir, "body.md");
-  copyFileSync(bodyFile, temporaryBodyFile);
-  appendFileSync(temporaryBodyFile, ["```diff", diff, "```"].join("\n"));
+  const body = makePullBody(diff);
 
   const pullRequestCmd = [
     "gh pr create",
     `--title="${title}"`,
     `--reviewer="${reviewer}"`,
-    `--body-file=${temporaryBodyFile}`,
+    `--body-file=-`,
     `--base="main"`,
     `--head="${releaseBranch}"`,
-    // `--repo="ddbeck/feature-set"`, // Uncomment this line to point this at a fork for testing
+    `--repo="${args.targetRepo}"`,
   ].join(" ");
-  run(pullRequestCmd);
+  execSync(pullRequestCmd, {
+    input: body,
+    stdio: ["pipe", "inherit", "inherit"],
+  });
+}
+
+function bumpVersion(semverlevel: typeof semverChoices): string {
+  const bumpCmd = `npm version --no-git-tag-version ${semverlevel}`;
+  logger.info("Bumping version number");
+  logger.debug(bumpCmd);
+  execSync(bumpCmd, { cwd: packages["web-features"], stdio: "inherit" });
+  const { version } = readPackageJSON(packages["web-features"]);
+
+  // Commit
+  logger.info("Committing version bump");
+  const commitMessage = `Increment ${semverlevel} version to v${version}`;
+  const commitCmd = `git commit --all --message="${commitMessage}"`;
+  run(commitCmd);
 
-  rmSync(temporaryDir, { recursive: true });
+  return version;
+}
+
+function makePullBody(diff: string) {
+  const bodyFile = fileURLToPath(
+    new URL("release-pull-description.md", import.meta.url)
+  );
+  const body = [
+    readFileSync(bodyFile, { encoding: "utf-8" }),
+    "```diff",
+    diff,
+    "```",
+  ].join("\n");
+  return body;
 }
 
 function update(args) {
-  // TODO: Rebase
-  // TODO: Run `npm run build`
-  // TODO: Commit results of `npm run build`
-  // TODO: Update PR diff
-  throw Error("Not implemented");
+  preflight({ expectedPull: args.pr, targetRepo: args.targetRepo });
+  build();
+
+  logger.verbose("Adding rebase-in-progress notice to PR description");
+  const { body } = JSON.parse(
+    execSync(
+      `gh pr view --repo="${args.targetRepo}" "${args.pr}" --json body`,
+      {
+        encoding: "utf-8",
+      }
+    )
+  );
+  const notice = "⛔️ Update in progress! ⛔️\n";
+  const editBodyCmd = `gh pr edit --repo="${args.targetRepo}" "${args.pr}" --body-file=-`;
+  execSync(editBodyCmd, {
+    input: [notice, body].join("\n\n"),
+    stdio: ["pipe", "inherit", "inherit"],
+  });
+
+  logger.verbose("Rebasing");
+  try {
+    run(`git rebase ${args.base}`);
+  } catch (err) {
+    logger.error("Rebasing failed. Abandoning PR.");
+    run(`git rebase --abort`);
+    run(
+      `gh pr comment --repo="${args.targetRepo}" "${args.pr}" --body="😱 Rebasing failed. Closing this PR. 😱"`
+    );
+    run(`gh pr close --repo="${args.targetRepo}" "${args.pr}"`);
+    process.exit(1);
```

Comment:
If the rebase fails, it closes the PR. My thinking here is that instead of being clever, we can wait for (or force) a new release PR instead.